edb/server/compiler/compiler.py

```diff
@@ -770,70 +764,220 @@ def _compile_and_apply_ddl_command(self, ctx: CompileContext, cmd):
+                )
+
+        elif isinstance(ql, qlast.CommitMigration):
+            mstate = current_tx.get_migration_state()
+            if mstate is None:
                 raise errors.QueryError(
-                    f'CREATE MIGRATION must be executed in a '
-                    f'transaction block')
+                    'unexpected POPULATE MIGRATION:'
+                    ' not currently in a migration block',
+                    context=ql.context,
+                )
 
-            assert self._std_schema is not None
-            cmd = s_ddl.compile_migration(
-                cmd,
-                self._std_schema,
-                current_tx.get_schema())
+            diff = s_ddl.delta_schemas(schema, mstate.target_schema)
+            if list(diff.get_subcommands()):
+                raise errors.QueryError(
+                    'cannot commit incomplete migration',
+                    hint=(
+                        'Please finish the migration by specifying the'
+                        ' remaining DDL operations or run POPULATE MIGRATION'
+                        ' to let the system populate the outstanding DDL'
+                        ' automatically.'
+                    ),
+                    context=ql.context,
+                )
+
+            last_migration = schema.get_last_migration()
+            if last_migration:
+                last_migration_ref = s_utils.name_to_ast_ref(
+                    last_migration.get_name(schema),
+                )
+            else:
+                last_migration_ref = None
+
+            create_migration = qlast.CreateMigration(
+                parent=last_migration_ref,
+                commands=mstate.current_ddl,
+                auto_diff=mstate.auto_diff,
+            )
+
+            current_tx.update_schema(mstate.initial_schema)
+            current_tx.update_migration_state(None)
+
+            ddl_query = self._compile_and_apply_ddl_stmt(
+                ctx,
+                create_migration,
+            )
+
+            if mstate.initial_savepoint:
+                savepoint_name = str(uuid.uuid4())
+                tx_cmd = qlast.DeclareSavepoint(name=savepoint_name)
+            else:
+                tx_cmd = qlast.CommitTransaction()
+
+            tx_query = self._compile_ql_transaction(ctx, tx_cmd)
+
+            query = dbstate.MigrationControlQuery(
+                sql=ddl_query.sql + tx_query.sql,
+                new_types=ddl_query.new_types,
+                action=dbstate.MigrationAction.COMMIT,
+                tx_action=tx_query.action,
+                cacheable=False,
+                modaliases=None,
+                single_unit=True,
+            )
+
+        elif isinstance(ql, qlast.AbortMigration):
+            mstate = current_tx.get_migration_state()
+            if mstate is None:
+                raise errors.QueryError(
+                    'unexpected ABORT MIGRATION:'
+                    ' not currently in a migration block',
```

Comment:
nit: I see you moving the space to the beginning of literal continuation line, whereas I (and I thought we all) used to keep it at the end. Not sure why we're switching the coding style.

Reply:
This was Łukasz's suggestion. When used consistently as a style, it makes it less likely that you forget the space at the end of the line or have it stripped by the editor accidentally (happened to me a few times).

Reply:
Well, now we have 5% of strings in new style and 95% in old. 🤷‍♂️ I wouldn't do this.